packages/cli/core/src/lib/utils/list.ts

```diff
@@ -1,3 +1,7 @@
 export function without<T>(array: T[], values: T[]): T[] {
   return array.filter((field) => !values.includes(field));
 }
+
+export function isSubset<T>(subset: T[], superset: T[]): boolean {
+  return without(subset, superset).length === 0;
```

Comment:
This is 'nice' from a DRY standpoint but could be more efficient.
Indeed this requires going through the whole `subset` element.

Currently, the method needs to do at least `n+1` comparisons, with `n` being the number of elements in the `subset`.
Indeed, we don't have a 'quick-exit' condition if the first element (or second, etc...) of `subset` is actually not in `superset`

So to fix that:

```suggestion
  return subset.every(element => superset.includes(element))
```

With this, at best, you have 1 operation to do: "is the first element of subset in superset? No, ok return false"

Given that your method requires subset to be the first param, you can also ask a precondition by checking the length of the two arrays:

```suggestion
  return subset.length > superset.length ? false : subset.every(element => superset.includes(element));
```

Reply:
> If we go that path, might as well compare real sets:
> 
> ```
>   if (new Set(subset).size > new Set(superset).size) {
>     return false;
>   }
>   return subset.every((element) => superset.includes(element));
> ```

I thought about it, but then I checked how the ctor of a Set work:
https://tc39.es/ecma262/multipage/keyed-collections.html#sec-set-iterable

You can see that, well, it will iterate through all values, so it would be costly to convert that as a set.

However, this made me think: the `if(subset.size>superset.size) return false` would not work: it assumes that every element of the subset and superset are unique, which we cannot know for sure:
```ts
const subset = [0,0,0,0,0,0,0,0]
const superset = [0,1,2,3]
// oops
```

The `subset.every(element => superset.includes(element))` is still OK tho.